command/agent/agent.go

```diff
@@ -733,6 +714,29 @@ func (a *Agent) purgeCheck(checkID types.CheckID) error {
 	return nil
 }
 
+// writeFileAtomic writes the given contents to a temporary file in the same
+// directory, does an fsync and then renames the file to its real path
+func writeFileAtomic(path string, contents []byte) error {
+	tempPath := path + ".tmp"
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil {
+		return err
+	}
+	fh, err := os.OpenFile(tempPath, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	if _, err := fh.Write(contents); err != nil {
```

Comment:
Oh that's a good point. We should probably even do a best-effort delete on the temp file if something goes wrong (do a deferred fn that we disarm at the end). We originally used a file name that was 1:1 with the service name but changed that to fix a race condition.